openai/api_resources/image.py

```diff
@@ -31,12 +34,20 @@ def create(
             organization=organization,
         )
 
-        _, api_version = cls._get_api_type_and_version(api_type, api_version)
+        api_type, api_version = cls._get_api_type_and_version(api_type, api_version)
 
         response, _, api_key = requestor.request(
-            "post", cls._get_url("generations"), params
+            "post", cls._get_url("generations", azure_action="submit", api_type=api_type, api_version=api_version), params
         )
 
+        if api_type in (util.ApiType.AZURE, util.ApiType.AZURE_AD):
+            requestor.api_base = "" # operation_location is a full url
+            response, _, api_key = requestor._poll(
+                "get", response.operation_location,
+                until=lambda response: response.data["status"] in ["succeeded"],
+                failed=lambda response: response.data['status'] in [ 'failed' ]
```

Comment:
Can we unify the formatting here? (`[ 'failed' ]` vs `["succeeded"]`). We'll add automatically linting/formatting to this repo soon